x-pack/test/functional/services/ml/data_visualizer_index_based.ts

```diff
@@ -231,5 +232,76 @@ export function MachineLearningDataVisualizerIndexBasedProvider({
         }
       );
     },
+
+    async assertRandomSamplingOptionsButtonExists() {
+      await testSubjects.existOrFail('dvRandomSamplerOptionsButton');
+    },
+
+    async assertRandomSamplingOption(
+      expectedOption:
+        | 'dvRandomSamplerOptionOnAutomatic'
+        | 'dvRandomSamplerOptionOnManual'
+        | 'dvRandomSamplerOptionOff',
+      expectedProbability?: number
+    ) {
+      await browser.pressKeys(browser.keys.ESCAPE);
+      await testSubjects.clickWhenNotDisabled('dvRandomSamplerOptionsButton');
+      await testSubjects.existOrFail('dvRandomSamplerOptionsPopover');
+
+      if (expectedOption === 'dvRandomSamplerOptionOff') {
+        await testSubjects.existOrFail('dvRandomSamplerOptionOff');
+        await testSubjects.missingOrFail('dvRandomSamplerProbabilityRange');
+        await testSubjects.missingOrFail('dvRandomSamplerAutomaticProbabilityMsg');
+      }
+
+      if (expectedOption === 'dvRandomSamplerOptionOnManual') {
+        await testSubjects.existOrFail('dvRandomSamplerOptionOnManual');
+        await testSubjects.existOrFail('dvRandomSamplerProbabilityRange');
+        if (expectedProbability !== undefined) {
+          const probability = await testSubjects.getAttribute(
+            'dvRandomSamplerProbabilityRange',
+            'value'
+          );
+          expect(probability).to.eql(
+            `${probability}`,
+            `Expected probability to be ${expectedProbability}, got ${probability}`
+          );
+        }
+      }
+
+      if (expectedOption === 'dvRandomSamplerOptionOnAutomatic') {
+        await testSubjects.existOrFail('dvRandomSamplerOptionOnAutomatic');
+        await testSubjects.existOrFail('dvRandomSamplerAutomaticProbabilityMsg');
+
+        if (expectedProbability !== undefined) {
+          const probabilityText = await testSubjects.getVisibleText(
+            'dvRandomSamplerAutomaticProbabilityMsg'
+          );
+          expect(probabilityText).to.contain(
+            `${expectedProbability}`,
+            `Expected probability to be ${expectedProbability}, got ${probabilityText}`
```

Comment:
Nit: we could make this assertion message a bit clearer, e.g.
```suggestion
          expect(probabilityText).to.contain(
            `${expectedProbability}`,
            `Expected probability text to contain ${expectedProbability}, got ${probabilityText}`
```

Reply:
Updated here [`d30e8ce` (#142278)](https://github.com/elastic/kibana/pull/142278/commits/d30e8ced311828876c68650bff5f5c18c88b6f29)